paprika/align.py

```diff
@@ -192,3 +192,158 @@ def offset_structure(structure, offset):
+    atom_mask : str
+        Selection of atom(s) if a particular subset is preferred to estimate the center.
+    dim_mask : list
+        A mask that will filter the dimensions to which the translation will be applied (default is [1,1,1]).
+
+    Returns
+    -------
+    structure : parmed.Structure
+        A molecular structure with the coordinates translated to the origin.
+
+    """
+    # Check if weight variable is properly chosen
+    if weight not in ["mass", "geo"]:
+        raise SystemExit('Error: "weight" must either be "mass" or "geo"')
+
+    # Dimension mask
+    if dim_mask is None:
+        dim_mask = [1, 1, 1]
+    elif len(dim_mask) != 3:
+        raise SystemExit(
+            'Error: "dim_mask" must be a list with 3 elements, e.g. [1,1,1]'
+        )
+    dim_mask = np.array(dim_mask)
+
+    # Atom coordinates and masses
+    if atom_mask is None:
+        coordinates = structure.coordinates
+        masses = np.asarray([atom.mass for atom in structure.atoms])
+    else:
+        coordinates = structure[atom_mask].coordinates
+        masses = np.asarray([atom.mass for atom in structure[atom_mask].atoms])
+
+    # Convert weight if geometric center
+    if weight == "geo":
+        masses[:] = 1.0
+
+    # Center of mass/geometry
+    centroid = pmd.geometry.center_of_mass(coordinates, masses)
+
+    # Translate coordinates
+    aligned_coords = np.empty_like(structure.coordinates)
+    for atom in range(len(structure.atoms)):
+        aligned_coords[atom] = structure.coordinates[atom] - centroid * dim_mask
+    structure.coordinates = aligned_coords
+
+    return structure
+
+
+def align_principal_axes(structure, atom_mask=None, princ_axis=None, v_axis=None):
+    """Aligns the a chosen principal axis of a system to a specified axis. 
+       This function is based on the method given in the link:
+
+        * https://www.ks.uiuc.edu/Research/vmd/script_library/scripts/orient/
+
+    Parameters
+    ----------
+    structure : parmed.Structure
+        Molecular structure containing coordinates.
+    atom_mask : str
+        A mask that filter specific atoms for calculating the moment of inertia.
+    princ_axis : int
+        Which principal axis to align? (choice: 1, 2 or 3, default: 1).
+    v_axis: list
+        The axis vector to align the system to (default: [0, 0, 1]).
+
+    Returns
+    -------
+    structure : parmed.Structure
+        A molecular structure with a principal axis aligned to a vector.
+
+    Examples
+    --------
+    The commands below mimics the example given in the link above for VMD.
+
+        >>> structure = align_principal_axes(structure, princ_axis=0, v_axis=[0,0,1])
+        >>> structure = align_principal_axes(structure, princ_axis=1, v_axis=[0,1,0])
+
+    """
+    # Check princ_axis
+    if princ_axis is None:
+        princ_axis = 1
+    elif princ_axis not in [0, 1, 2]:
+        raise SystemExit(
```

Comment:
Same as above w.r.t `SystemExit`. 

For these kinds of checks you can also look into Enums. I learned about them from Simon and just started using them at the end of last year.

Reply:
Changed to Exception